### ISSUE TITLE
thumbnails for folders?

### ISSUE THREAD

**Issue by mrjohnpoz** [sjarvela/kloudspeaker#6]
Is it possible to enable thumbnails and if in large icon view display specific image based on folder name as icon vs generic folder icon?

**Comment by sjarvela** [comment]
This could be possible like this:
- if folder contains file "_thumb.png", use it
- ignore "_thumb.png" files on any folder so they won't be visible in listings

The problem could be, if file does not exist, how to fall back to default icon. At the moment, if thumbnails are enabled, it generates thumbnail url for the image, and it must return an image or the browser will show empty element. And specifying fallback image in server side is not an option, because the default image is matter of client side CSS.

**Comment by luke7263** [comment]
Samuli,  I don't understand .. You mean you just have to generate a "_thumb.png" in that directory ? Don't work ..:(

**Comment by sjarvela** [comment]
No, I meant it _could_ work like that if I added the support, but it does not exist at the moment. But like I said, there are flaws in that idea...

**Comment by nnmalex** [comment]
Maybe you could add a transparent layer over the folder and show image there if _thumb.png is present. Otherwise nothing will be shown and you'll see the default folder icon underneath?

**Comment by nnmalex** [comment]
By the way there are thumbnails for file types but CSS contains description only for PDF files. Adding more types to CSS helps but it will make updates harder. Is it possible that you would enable more file types by default?

**Comment by sjarvela** [comment]
It is possible of course, and have had intention to do so. If you have a suggestion for mime type icons you have added, send them to me and I'll add it to the release.

**Comment by nnmalex** [comment]
I have used those icons already provided with the distro. I have only edited your CSS (basic copy/paste of existed pdf handling, choosing appropriate icon from /css/images/mimetypes*) for the below file types:
mov, mp4, avi, zip, ppt, pptx, doc, docx, xls, xlsx

**Comment by sjarvela** [comment]
Exactly, that was my assumption. I just meant that if you have that already done, I could just insert it to the released version :) any help is appreciated as my time is very limited

**Comment by nnmalex** [comment]
Ah, OK. :)

---First part of CSS---
/\* Original */
.kloudspeaker-filelist-item.item-type-pdf .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/pdf.png) no-repeat;
}

/\* Added */
.kloudspeaker-filelist-item.item-type-zip .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/zip.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-mp4 .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/video_mp4.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-mov .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/video_mp4.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-ppt .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/application_vnd.ms_powerpoint.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-pptx .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/application_vnd.ms_powerpoint.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-doc .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/application_vnd.ms_word.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-docx .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/application_vnd.ms_word.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-xls .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/x_office_spreadsheet.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-xlsx .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/x_office_spreadsheet.png) no-repeat;
}

.kloudspeaker-filelist-item.item-type-psd .kloudspeaker-filelist-item-icon {
    background: url(images/mimetypes/image_x_psd.png) no-repeat;
}

/\* End of added */

---Second part of CSS---

/\* Original */
.kloudspeaker-iconview-item.item-type-pdf .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/pdf.png);
}

/\* Added */
.kloudspeaker-iconview-item.item-type-zip .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/zip.png);
}

.kloudspeaker-iconview-item.item-type-mp4 .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/video_mp4.png);
}

.kloudspeaker-iconview-item.item-type-mov .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/video_mp4.png);
}

.kloudspeaker-iconview-item.item-type-ppt .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_powerpoint.png);
}

.kloudspeaker-iconview-item.item-type-pptx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_powerpoint.png);
}

.kloudspeaker-iconview-item.item-type-doc .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_word.png);
}

.kloudspeaker-iconview-item.item-type-docx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_word.png);
}

.kloudspeaker-iconview-item.item-type-xls .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_excel.png);
}

.kloudspeaker-iconview-item.item-type-xlsx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/application_vnd.ms_excel.png);
}

.kloudspeaker-iconview-item.item-type-psd .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes128/image_x_psd.png);
}

/\* End of added _/
/_ Original */
.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-pdf .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/pdf.png);
}

/\* Added */
.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-zip .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/zip.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-mp4 .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/video_mp4.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-mov .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/video_mp4.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-ppt .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_powerpoint.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-pptx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_powerpoint.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-doc .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_word.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-docx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_word.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-xls .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_excel.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-xlsx .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/application_vnd.ms_excel.png);
}

.kloudspeaker-iconview.iconview-small .kloudspeaker-iconview-item.item-type-psd .kloudspeaker-iconview-item-icon {
    background-image: url(images/mimetypes64/image_x_psd.png);
}

/\* End of added */

**Comment by nnmalex** [comment]
Thumbnails won't appear even if properly described in CSS if file extension is written in capital letters. Is it possible to fix this?

**Comment by sjarvela** [comment]
Oh yes, I'll just force the css class to lowercase.

**Comment by sjarvela** [comment]
Committed the fix, will be in next release.
